dm/master/server.go

```diff
@@ -1707,53 +1824,114 @@ var (
 	retryInterval = time.Second
 )
 
-func (s *Server) waitOperationOk(ctx context.Context, cli pb.WorkerClient, name string, opLogID int64) error {
-	request := &pb.QueryTaskOperationRequest{
-		Name:  name,
-		LogID: opLogID,
+func (s *Server) waitOperationOk(ctx context.Context, cli workerrpc.Client, name string, opLogID int64) error {
+	req := &workerrpc.Request{
+		Type: workerrpc.CmdQueryTaskOperation,
+		QueryTaskOperation: &pb.QueryTaskOperationRequest{
+			Name:  name,
+			LogID: opLogID,
+		},
 	}
 
 	for num := 0; num < maxRetryNum; num++ {
-		res, err := cli.QueryTaskOperation(ctx, request)
+		resp, err := cli.SendRequest(ctx, req, s.cfg.RPCTimeout)
 		if err != nil {
 			log.Errorf("fail to query task operation %v", err)
-		} else if res.Log == nil {
-			return errors.Errorf("operation %d of task %s not found, please execute `query-status` to check status", opLogID, name)
-		} else if res.Log.Success {
-			return nil
-		} else if len(res.Log.Message) != 0 {
-			return errors.New(res.Log.Message)
+		} else {
+			respLog := resp.QueryTaskOperation.Log
+			if respLog == nil {
+				return errors.Errorf("operation %d of task %s not found, please execute `query-status` to check status", opLogID, name)
+			} else if respLog.Success {
+				return nil
+			} else if len(respLog.Message) != 0 {
+				return errors.New(respLog.Message)
+			}
 		}
 
-		log.Infof("wait task %s op log %d, current result %+v", name, opLogID, res)
+		log.Infof("wait task %s op log %d, current result %+v", name, opLogID, resp.QueryTaskOperation)
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
 		case <-time.After(retryInterval):
 		}
-
 	}
 
 	return errors.New("request is timeout, but request may be successful, please execute `query-status` to check status")
 }
 
-func (s *Server) handleOperationResult(ctx context.Context, cli pb.WorkerClient, name string, err error, response *pb.OperateSubTaskResponse) *pb.OperateSubTaskResponse {
+func (s *Server) handleOperationResult(ctx context.Context, cli workerrpc.Client, name string, err error, resp *workerrpc.Response) *pb.OperateSubTaskResponse {
 	if err != nil {
 		return &pb.OperateSubTaskResponse{
-			Meta: &pb.CommonWorkerResponse{
-				Result: false,
-				Msg:    errors.ErrorStack(err),
-			},
+			Meta: errorCommonWorkerResponse(errors.ErrorStack(err), ""),
+		}
+	}
+	response := &pb.OperateSubTaskResponse{}
+	switch resp.Type {
+	case workerrpc.CmdStartSubTask:
+		response = resp.StartSubTask
+	case workerrpc.CmdOperateSubTask:
+		response = resp.OperateSubTask
+	case workerrpc.CmdUpdateSubTask:
+		response = resp.UpdateSubTask
+	default:
+		return &pb.OperateSubTaskResponse{
+			Meta: errorCommonWorkerResponse(fmt.Sprintf("invalid operate task type %v", resp.Type), ""),
```

Comment:
`response.Meta = xxx`？

Reply:
this is an unexpected branch, should not ever happen

Reply:
if so, it's better to keep response